build_helper.py

```diff
@@ -0,0 +1,195 @@
+         'platform': DEFAULT_ANACONDA_ARCH,
+         'install': 'conda build .',
+         'user': 'user'
+    }
+
+def cmd(args, cwd='.'):
+    proc = sp.Popen(args, stdout=sp.PIPE,
+                    stderr=sp.PIPE, cwd=cwd)
+    while proc.poll is None:
+        time.sleep(1)
+    out = proc.stdout.read().decode()
+    err = proc.stderr.read().decode()
+    return proc.returncode, out, err
+
+def raise_or_not(args, msg):
+    if args.exception_action == 'raise':
+        raise ValueError(msg)
+    print(msg, file=sys.stderr)
+
+def choose_engine_install_channels(meta, args):
+    reqs = meta.get('requirements', {})
+    build_reqs = reqs.get('build', None)
+    run_reqs = reqs.get('run', None)
+    if build_reqs is None:
+        build_reqs = []
+    if run_reqs is None:
+        run_reqs = []
+    b = [_.strip().lower() for _ in build_reqs + run_reqs]
+    engine, install_channels = [],['defaults']
+    if 'python' in b:
+        engine = ['python']
+        install_channels += ['python']
+    if 'r' in b:
+        engine += ['r']
+        install_channels += ['r']
+    return engine, install_channels
+
+create_package_cmd = 'anaconda package --create {username}/{package}'
+build_submit_cmd = 'anaconda build submit {package_file} --queue {queue} --channel {channel}'
+
+def submit_package(args, top_dir, package_name):
+    with open(join(top_dir, package_name, '.binstar.yml'), 'r') as f:
+        binstar_yml = yaml.load(f.read())
+    create = create_package_cmd.format(username=binstar_yml['user'],
+                                       package=package_name).split()
+    fil = join(top_dir, package_name)
+    build = build_submit_cmd.format(package_file=fil,
+                                    queue=args.queue,
+                                    channel=args.channel).split()
+    returncode2, out2, err2 = cmd(create, cwd=top_dir)
+    if returncode2:
+        raise ValueError('{} failed with {}, {}'.format(create, out2, err2))
+    returncode3, out3, err3 = cmd(build, cwd=top_dir)
+    print('Submitted:',build,'\n\n', out3 + '\n\n' + err3)
+    if returncode3:
+        raise ValueError('{} failed with {}, {}'.format(build, out3, err3))
+
+def on_each_package(args, top_dir, package_name):
+    if args.action == 'edit':
+        meta_file = join(top_dir, package_name, 'meta.yaml')
+        if not exists(meta_file):
+            has_subdirs = False
+            for try_levels in range(1,10):
+                parts = (top_dir, package_name,) + ('*',) * try_levels
+                parts = parts + ('meta.yaml',)
+                if glob.glob(join(*parts)):
+                    has_subdirs = True
+
+            if not has_subdirs:
+                msg = "Package {} in top_dir {} has no meta.yaml"
+                msg = msg.format(package_name, top_dir)
+                raise ValueError(msg)
+            else:
+                return
+        with open(meta_file, 'r') as f:
+            meta = yaml.load(f.read())
```

Comment:
You should be using conda's loading functions for meta.yaml, since meta.yaml is not standard yaml.